src/core/tsi/ssl_transport_security.cc

```diff
@@ -1952,6 +1989,22 @@ tsi_result tsi_create_ssl_client_handshaker_factory_with_options(
     SSL_CTX_set_session_cache_mode(ssl_context, SSL_SESS_CACHE_CLIENT);
   }
 
+  if (options->key_logger != nullptr) {
+    // Unref is manually called on factory destruction
+    impl->key_logger =
+        reinterpret_cast<tsi::TlsKeyLogger*>(options->key_logger)->Ref();
+    if (options->session_cache == nullptr) {
+      // Need to set factory at g_ssl_ctx_ex_factory_index
+      SSL_CTX_set_ex_data(ssl_context, g_ssl_ctx_ex_factory_index, impl);
+    }  // Otherwise it is already set
+
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
```

Comment:
Is it possible to move the macros defined in `ssl_key_logging.cc` to this file, so that we don't need any additional macros in `ssl_key_logging.cc`(I just wanted to reduce the number of macros if possible, since they affect readability)?
I think ideally speaking, `ssl_key_logging.cc` doesn't need to contain any macros, since it contains some self-contained classes that not restricted to specific OpenSSL versions. It was the caller of these classes(ssl_transport_security.cc) that can only work for certain OpenSSL versions.

Reply:
Done.